extensions/timeseries/src/main/java/ai/djl/timeseries/dataset/M5Forecast.java

```diff
@@ -0,0 +1,246 @@
+
+import ai.djl.basicdataset.tabular.utils.Feature;
+import ai.djl.basicdataset.tabular.utils.Featurizers;
+import ai.djl.repository.MRL;
+import ai.djl.repository.Repository;
+import ai.djl.repository.zoo.DefaultModelZoo;
+import ai.djl.util.JsonUtils;
+import ai.djl.util.Progress;
+
+import org.apache.commons.csv.CSVFormat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * M5 Forecasting - Accuracy from <a
+ * href="https://www.kaggle.com/competitions/m5-forecasting-accuracy">https://www.kaggle.com/competitions/m5-forecasting-accuracy</a>
+ *
+ * <p>To improve the model performance, we coarse graining target of the dataset by summing the sale
+ * amount every seven days. And set the column names of sum as 'w_i'. This can reduce occurrence of
+ * invalid values 0 and reduce the noise learned by model.
+ */
+public class M5Forecast extends CsvTimeSeriesDataset {
+
+    private Usage usage;
+    private MRL mrl;
+    private boolean prepared;
+    private Path root;
+    private List<Integer> cardinality;
+
+    /**
+     * Creates a new instance of {@link M5Forecast} with the given necessary configurations.
+     *
+     * @param builder a builder with the necessary configurations
+     */
+    protected M5Forecast(Builder builder) {
+        super(builder);
+        usage = builder.usage;
+        String path = builder.repository.getBaseUri().toString();
+        mrl = MRL.undefined(builder.repository, DefaultModelZoo.GROUP_ID, path);
+        root = Paths.get(mrl.getRepository().getBaseUri());
+        cardinality = builder.cardinality;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void prepare(Progress progress) throws IOException {
+        if (prepared) {
+            return;
+        }
+
+        mrl.prepare(null, progress);
+        Path csvFile = root.resolve(getUsagePath(usage));
+
+        csvUrl = csvFile.toUri().toURL();
+        super.prepare(progress);
+        prepared = true;
+    }
+
+    /**
+     * Return the cardinality of the dataset.
+     *
+     * @return the cardinality of the dataset
+     */
+    public List<Integer> getCardinality() {
+        return cardinality;
+    }
+
+    /**
+     * Creates a builder to build a {@link M5Forecast}.
+     *
+     * @return a new builder
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private String getUsagePath(Usage usage) {
+        // We coarse graining the data by summing the sale amount every 7 days and rename the .csv
+        // file as 'weekly_***'
+        String usagePath;
+        switch (usage) {
+            case TRAIN:
+                usagePath = "weekly_sales_train_validation.csv";
```

Comment:
Is there is a script to generate this file? I think we should upload this file to S3 as dataset

Reply:
Yes. This is a python script, which transforms the daily time series data to weekly data.